src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2596,21 +2597,28 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatement(SyntaxKind par
             SyntaxToken identifierOrThisOpt,
             TypeParameterListSyntax typeParameterListOpt)
         {
-            if (modifiers.Any(SyntaxKind.AsyncKeyword)) return false;
             if (type.Kind != SyntaxKind.IdentifierName) return false;
-            if ((((IdentifierNameSyntax)type).Identifier).ContextualKind != SyntaxKind.AsyncKeyword) return false;
             if (identifierOrThisOpt.Kind != SyntaxKind.IdentifierToken) return false;
-            modifiers.Add(ConvertToKeyword(((IdentifierNameSyntax)type).Identifier));
-            SimpleNameSyntax newType = typeParameterListOpt == null
-                ? (SimpleNameSyntax)_syntaxFactory.IdentifierName(identifierOrThisOpt)
-                : _syntaxFactory.GenericName(identifierOrThisOpt, TypeArgumentFromTypeParameters(typeParameterListOpt));
-            type = (explicitInterfaceOpt == null)
-                ? (TypeSyntax)newType
-                : _syntaxFactory.QualifiedName(explicitInterfaceOpt.Name, explicitInterfaceOpt.DotToken, newType);
-            explicitInterfaceOpt = null;
-            identifierOrThisOpt = default(SyntaxToken);
-            typeParameterListOpt = default(TypeParameterListSyntax);
-            return true;
+
+            var identifier = ((IdentifierNameSyntax)type).Identifier;
+            var contextualKind = identifier.ContextualKind;
+            if ((contextualKind == SyntaxKind.AsyncKeyword || contextualKind == SyntaxKind.ReplaceKeyword) &&
+                !modifiers.Any(contextualKind))
+            {
+                modifiers.Add(ConvertToKeyword(identifier));
+                SimpleNameSyntax newType = typeParameterListOpt == null
+                    ? (SimpleNameSyntax)_syntaxFactory.IdentifierName(identifierOrThisOpt)
+                    : _syntaxFactory.GenericName(identifierOrThisOpt, TypeArgumentFromTypeParameters(typeParameterListOpt));
+                type = (explicitInterfaceOpt == null)
+                    ? (TypeSyntax)newType
+                    : _syntaxFactory.QualifiedName(explicitInterfaceOpt.Name, explicitInterfaceOpt.DotToken, newType);
+                explicitInterfaceOpt = null;
+                identifierOrThisOpt = default(SyntaxToken);
+                typeParameterListOpt = default(TypeParameterListSyntax);
+                return true;
+            }
+
+            return false;
```

Comment:
Not a blocker. Consider using the negative condition and returning false right then and there.